app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -515,6 +515,8 @@ def vm_hardware_guest_devices(persister_hardware, vm, addresses, host)
       network = mac && networks.present? ? networks[mac] : nil
 
       vnic_profile_id = nic.dig(:vnic_profile, :id)
+      next if vnic_profile_id.nil?
```

Comment:
@borod108 if this happens I wasn't sure if it was better to skip the NIC completely, or just not lookup the switch and lan below

Reply:
I think it is not.